fe/fe-core/src/main/java/org/apache/doris/transaction/DatabaseTransactionMgr.java

```diff
@@ -342,6 +339,344 @@ public void updateDatabaseUsedQuotaData(long usedQuotaDataBytes) {
         this.usedQuotaDataBytes = usedQuotaDataBytes;
     }
 
+    public void preCommitTransaction(List<Table> tableList, long transactionId, List<TabletCommitInfo> tabletCommitInfos,
```

Comment:
Most part of this method is same as `commitTransaction`. Can we merge these 2 methods?
I think the only difference it to set the txn state to PRECOMMITTED or COMMITTED.